### PR TITLE
Card #9

### PR DESCRIPTION
**Pull request by sonalik-cuelogic**
Card Link: https://trello.com/c/mvB76SGS
Implemented As an empolyee, I can view my own and my subordinate leaves.

### CODE REVIEW

File: leave_mgmt/templates/leave_mgmt/leave_view.html [hunk 1]
```diff
@@ -1,19 +1,95 @@
...
 
 {% block scripts_extend %}
-    <script type="text/javascript" src="{% static 'js/leave_view.js' %}"></script>
-{% endblock scripts_extend %}
+<script src="//code.jquery.com/jquery-1.10.2.js"></script>
+<script src="//code.jquery.com/ui/1.11.4/jquery-ui.js"></script>
+<script type="text/javascript" src="{% static 'js/config.js' %}"></script>
+<script type="text/javascript" src="{% static 'js/leave_view.js' %}"></script>
+<script type="text/javascript" src="{% static 'js/bootstrap-multiselect.js' %}"></script>
+<script type="text/javascript" src="{% static 'js/jquery.dataTables.min.js' %}"></script>
```
Comment (aloka-cuelogic) [inline leave_mgmt/templates/leave_mgmt/leave_view.html]: Where is this js file located ?
This file is not present in system ?
remove references of same.

File: leave_mgmt/templates/leave_mgmt/leave_view.html [hunk 2]
```diff
@@ -1,19 +1,95 @@
...
+<div class="alert alert-danger" id="difference_error">
+</div>
+
+<div class="row content_header">
+    <div class="col-sm-2 multiselect">
+        <select id="id_Team_Members" class="btn btn-default" onchange='populate_member_results()' name="members_filter" multiple="multiple">
+        {% for member in team_members %}
+        <option class="btn-default" value="{{member.id}}"
+            {% for employee in members_filter %}
+                {% ifequal employee.id member.id %}
```
Comment (aloka-cuelogic) [inline leave_mgmt/templates/leave_mgmt/leave_view.html]: What's the use of this condition ?
If you are using jQuery for getting member details?
This code for pre-selecting members is not required.

File: leave_mgmt/templates/leave_mgmt/leave_view.html [hunk 3]
```diff
@@ -1,19 +1,95 @@
 {% extends 'base.html' %}
-
 {% load staticfiles %}
 
 {% block scripts_extend %}
-    <script type="text/javascript" src="{% static 'js/leave_view.js' %}"></script>
-{% endblock scripts_extend %}
+<script src="//code.jquery.com/jquery-1.10.2.js"></script>
```
Comment (aloka-cuelogic) [inline leave_mgmt/templates/leave_mgmt/leave_view.html]: Please download these files and keep a local copy of same in static/js

File: leave_mgmt/templates/leave_mgmt/leave_view.html [hunk 4]
```diff
@@ -1,19 +1,95 @@
...
+<script src="//code.jquery.com/ui/1.11.4/jquery-ui.js"></script>
```
Comment (aloka-cuelogic) [inline leave_mgmt/templates/leave_mgmt/leave_view.html]: Same as above

File: leave_mgmt/templates/leave_mgmt/leave_view.html (hunk 3 above)
Comment (ananthu-nair707) [inline leave_mgmt/templates/leave_mgmt/leave_view.html]: http://webmasters.stackexchange.com/a/13932

Its Ok to use a CDN if its available.Actually its better if a CDN is available for any content resource. Please refer above link.

File: leave_mgmt/views.py [hunk 5]
```diff
@@ -1,9 +1,122 @@
...
 import datetime
-from employee_details.models import User_Profile
+import json
+
+from calendar import month_name
+from collections import OrderedDict
+
+from django.http import HttpResponse
+from django.shortcuts import render
+from datetime import date, timedelta
```
Comment (aloka-cuelogic) [inline leave_mgmt/views.py]: Either used import datetime Line 1
or use
from datetime import date, timedelta
You can use datetime.timedelta

importing both are not required.

File: static/js/leave_view.js [hunk 6]
```diff
@@ -1,3 +1,209 @@
...
+    var from_date = new Date(start_date[2], start_date[1] - 1, start_date[0]);
+    var to_date = new Date(end_date[2], end_date[1] - 1, end_date[0]);
+    var timeDiff = Math.abs(to_date.getTime() - from_date.getTime());
+    var diffDays = Math.ceil(timeDiff / (1000 * 3600 * 24));
+    if (diffDays > 30 && diffDays < 122 && to_date > from_date) {
+        selected_month = 0;
+        data = fetch_result(selected_month);
+        leaves_table_view(selected_month, data);
+    } else {
+        $("#difference_error").text("Select Start Date and End date correctly...");
```
Comment (aloka-cuelogic) [inline static/js/leave_view.js]: Please give a specific message.
like for less than 30 days give a message like to_date and from_date should have a minimum difference of 30 days.

**Comment by aloka-cuelogic** [comment]
@sonalik-cuelogic
 
Please implement the review comments.

File: leave_mgmt/templates/leave_mgmt/leave_view.html (hunk 3 above)
Comment (aloka-cuelogic) [inline leave_mgmt/templates/leave_mgmt/leave_view.html]: We are planning to make some changes in jquery-1.10.2.js, also if the site is only working in Intranet, so if the internet is not available the code will fail.(I think the site is hosted locally).

Additionally I agree that there are certain advantage of using CDN but additionally

Local has following advantage:

development without being connected to the net is possible
can still get some performance benefits by gzipping, in addition to minifying.

Comment (aloka-cuelogic) [inline leave_mgmt/templates/leave_mgmt/leave_view.html]: Some Other pitfall of CDN



    If you only include content from your own server, that's one server that needs to be running—and not blocked by firewalls etc—to make your site work. Pull script from a third party and now that's two servers that need to be running and unblocked to make your site work.

    Any site you pull <script> from can completely control the user's experience on your site. If Google were feeling evil they could put something in their copy of jQuery to log your keypresses, steal personal information from the page you're on to tie into their web tracking database, make you post “I love Google!” comments to every form, and so on.

Google probably aren't actually going to do that, but it's a factor that's out of your control, and certainly something to worry about with other script-hosting services. There have been incidents before where stats scripts have been compromised with malware loaders.

Before including any script from a third party—even on one single page of your site—you must 100% trust them with all user-accessible functionality visible on that hostname (including web-facing admin functions).

**Comment by sonalik-cuelogic** [comment]
Implemented review Comments

File: leave_mgmt/views.py [hunk 7]
```diff
@@ -1,9 +1,121 @@
...
+    leaves_dict = OrderedDict()
+
+    data = request.GET
+    members_filter_id = data.getlist('members_filter[]')
+    duration = data.get('duration')
+    start_date = data.get('start_date')
+    end_date = data.get('end_date')
+
+    members_list = [User_Profile.objects.get(user=request.user)]
+    print members_list
```
Comment (aloka-cuelogic) [inline leave_mgmt/views.py]: remove print statement

File: leave_mgmt/views.py [hunk 8]
```diff
@@ -1,9 +1,121 @@
...
+            members_list.append(User_Profile.objects.get(id=member_id))
+
+    # This has been kept to maintain an order of
+    # displaying logged in user followed by user team member
+    user_data = OrderedDict()
+
+    for member in members_list:
+        user_data[member.id] = member.user.first_name + \
+            ' ' + member.user.last_name
+    print user_data
```
Comment (aloka-cuelogic) [inline leave_mgmt/views.py]: remove Print statement

**Comment by aloka-cuelogic** [comment]
Reviewed

**Comment by sonalik-cuelogic** [comment]
@aloka-cuelogic 

As discussed have implemented multiple hierarchy support.

**Comment by aloka-cuelogic** [comment]
@onkartibe 

I am not having merge right for this repo.
Have verified the PR.
Tested all the scenarios all things are working as expected.

@sonalik-cuelogic 
Proceed with merge of PR and continue next task.

**Pull request merged**